samcli/cli/main.py

```diff
@@ -117,6 +117,14 @@ def cli(ctx):
     import atexit
     from samcli.lib.telemetry.metric import send_installed_metric, emit_all_metrics
 
+    # if development version of SAM CLI is used, attach module proxy
+    # to catch missing configuration for dynamic/hidden imports
+    if ctx and ctx.command_path == "samdev":
```

Comment:
Is there any risk that this can impact the production cli? 

General things look fine but just thinking of worse cases here. The way this is done it will package the "samdev" only things within the production CLI. 

Just want to ensure we thought this through.